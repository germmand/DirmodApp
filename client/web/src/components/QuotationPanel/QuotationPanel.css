.panel {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    width: 100%;
    padding: 10px;
}

.text-bold {
    font-size: 30px;
    font-weight: bold;
}

.flag-code-container {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-around;
}
