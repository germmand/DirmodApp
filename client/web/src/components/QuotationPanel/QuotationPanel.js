import React from 'react';
import PropTypes from 'prop-types';

import Card from 'react-bootstrap/Card';
import Spinner from 'react-bootstrap/Spinner';

import FlagIcon from '../FlagIcon';

import './QuotationPanel.css';

const QuotationPanel = ({ code, symbol, quotation, flagSize, isRefreshing }) => (
  <Card body>
    <div className="panel">
      <div className="flag-code-container">
        <FlagIcon code={code} size={flagSize} />
        <span className="text-bold">
          &nbsp;
          { code }
        </span>
      </div>
      { !isRefreshing ? (
        <div>
          <span className="text-bold">{ symbol }</span>
          <span className="text-bold">
            {' '}
            { quotation }
          </span>
        </div>
        ) : (
          <Spinner animation="border" />
        )}
    </div>
  </Card>
);

QuotationPanel.propTypes = {
  code: PropTypes.string.isRequired,
  symbol: PropTypes.string.isRequired,
  quotation: PropTypes.number.isRequired,
  isRefreshing: PropTypes.bool.isRequired,
  flagSize: PropTypes.number,
};

QuotationPanel.defaultProps = {
  flagSize: 55,
};

export default QuotationPanel;