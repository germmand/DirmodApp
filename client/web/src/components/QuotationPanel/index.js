import QuotationPanel from './QuotationPanel';

export default QuotationPanel;