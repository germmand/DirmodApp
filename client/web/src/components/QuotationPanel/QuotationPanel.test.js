import React from 'react';
import renderer from 'react-test-renderer';

import { shallow } from 'enzyme';

import QuotationPanel from './QuotationPanel';

describe('QuotationPanel Component', () => {
  it('should match snapshot', () => {
    const props = {
      code: 'USD',
      symbol: '$',
      quotation: 45.55,
      isRefreshing: false,
    };
    const component = renderer.create(
      <QuotationPanel {...props} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render spinner when isRefreshing equals true', () => {
    const props = {
      code: 'USD',
      symbol: '$',
      quotation: 45.55,
      isRefreshing: false,
    };
    const wrapper = shallow(
      <QuotationPanel {...props} />
    );
    expect(wrapper.find('Spinner')).toHaveLength(0);
    wrapper.setProps({
      isRefreshing: true,
    });
    expect(wrapper.find('Spinner')).toHaveLength(1);
  });
});