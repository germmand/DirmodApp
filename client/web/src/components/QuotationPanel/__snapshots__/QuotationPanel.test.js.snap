// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuotationPanel Component should match snapshot 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <div
      className="panel"
    >
      <div
        className="flag-code-container"
      >
        <img
          alt="USD"
          src="US.svg"
          style={
            Object {
              "display": "inline-block",
              "height": "55px",
              "width": "55px",
            }
          }
        />
        <span
          className="text-bold"
        >
          Â 
          USD
        </span>
      </div>
      <div>
        <span
          className="text-bold"
        >
          $
        </span>
        <span
          className="text-bold"
        >
           
          45.55
        </span>
      </div>
    </div>
  </div>
</div>
`;
